hub/tools/tagger.js

```diff
@@ -0,0 +1,271 @@
+          // Sub-tag, need to show parent tag
+          path = `${tag.level1}/`;
+        }
+
+        // Tag contains part of search term
+        if (offset !== -1) {
+          html += `<span class="tag" data-path="${path}"
+                        data-filter="${tag.filter}"
+                        data-category="${tag.category}"
+                        data-name="${tag.name}">`;
+          if (path.length) {
+            html += `<span class="tag--path">${path}</span>`;
+          }
+          html += `<span class="tag--name">${tag.name}</span>`;
+          html += '</span>';
+        }
+      }
+    });
+    html += '</div>';
+  });
+
+  document.getElementById('results').innerHTML = html;
+}
+
+function updateSelection() {
+  const filters = Object.keys(taxonomy.getFilters());
+  const categories = Object.keys(taxonomy.getCategories());
+  const selection = document.getElementById('selection');
+  const selectionValues = {};
+  let html = '';
+
+  selectedTags.sort((a, b) => {
+    // Compare by category first
+    const categoryPositionItemA = categories.indexOf(a.category);
+    const categoryPositionItemB = categories.indexOf(b.category);
+    if (categoryPositionItemA < categoryPositionItemB) {
+      return -1;
+    }
+    if (categoryPositionItemA > categoryPositionItemB) {
+      return 1;
+    }
+
+    // Compare by filter
+    const filterPositionItemA = filters.indexOf(a.filter);
+    const filterPositionItemB = filters.indexOf(b.filter);
+    if (filterPositionItemA < filterPositionItemB) {
+      return -1;
+    }
+    if (filterPositionItemA > filterPositionItemB) {
+      return 1;
+    }
+
+    // Compare by path
+    if (a.path < b.path) {
+      return -1;
+    }
+
+    if (a.path > b.path) {
+      return 1;
+    }
+
+    // Compare by name
+    if (a.name < b.name) {
+      return -1;
+    }
+    if (a.name > b.name) {
+      return 1;
+    }
+
+    return -1;
+  });
+
+  selectedTags.forEach((tag) => {
+    html += `<span class="selectedTag" data-filter="${tag.filter}">`;
+    if (tag.path.length) {
+      html += `<span class="tag--path">${tag.path}</span>`;
+    }
+    html += `<span class="tag--name">${tag.name}</span>`;
+    html += '</span>';
+
+    // Add to selected category input
+    if (!Object.prototype.hasOwnProperty.call(selectionValues, tag.category)) {
+      selectionValues[tag.category] = [];
+    }
+    selectionValues[tag.category].push(tag.name);
+  });
+
+  // Build the text to be copied and button actions
+  html += '<div class="copy-buttons">Copy tags for:';
+  Object.entries(selectionValues).forEach(([key, value]) => {
+    html += `<button data-target="selection-${key}" class="copy-button">${key}</button>`;
+    html += `<input type="text" id="selection-${key}" value="${value.join(',')}" class="notInViewport" />`;
+  });
+  html += '<div>';
+
+  document.addEventListener('click', (event) => {
+    const element = event.target;
+    if (element.classList.contains('copy-button')) {
```

Comment:
I remember buttons sometimes act weird when you manage to click on the text itself. May I suggest you do something like `if (element.closest('.copy-button')) {`. This way the button is targeted no matter which part of it is clicked.